File: swift/DesignPattern.playground/Pages/SimpleFactoryPattern.xcplaygroundpage/Sources/BeverageShop.swift
import Foundation

public class BeverageShop {
    
    private let factory: BeverageFactory
    
    public init(factory: BeverageFactory) {
        self.factory = factory
    }
    
    public func order(beverageName: String) -> Beverage? {
        let beverage: Beverage? = factory.createBeverage(beverageName: beverageName)
        
        beverage?.addSuger(level: 5)
        beverage?.addIce(level: 5)
        beverage?.shake()
        beverage?.packageUp()
        
        return beverage
    }
}

File: kotlin/DesignPattern/src/main/kotlin/factorymethodpattern/source/BeverageShop.kt
package factorymethodpattern.source

class BeverageShop(private val factory: BeverageFactory) {

    fun order(beverageName: String): Beverage? {
        val beverage: Beverage? = factory.createBeverage(beverageName)

        beverage?.addSuger(5)
        beverage?.addIce(5)
        beverage?.shake()
        beverage?.packageUp()

        return  beverage
    }
}